playbooks/common/openshift-cluster/upgrades/v3_5/validator.yml

```diff
@@ -0,0 +1,51 @@
+---
+###############################################################################
+# Pre upgrade checks for known data problems, if this playbook fails you should
+# contact support. If you're not supported contact users@lists.openshift.com
+#
+# oc_objectvalidator provides these two checks
+# 1 - SDN Data issues, never seen in the wild but known possible due to code audits
+#     https://github.com/openshift/origin/issues/12697
+# 2 - Namespace protections, https://bugzilla.redhat.com/show_bug.cgi?id=1428934
+#
+###############################################################################
+- name: Verify 3.5 specific upgrade checks
+  hosts: oo_first_master
+  roles:
+  - { role: lib_openshift }
+  tasks:
+  - name: Check for invalid namespaces and SDN errors
+    oc_objectvalidator:
+
+  # What's all this PetSet business about?
+  #
+  # 'PetSets' were ALPHA resources in Kube <= 3.4. In >= 3.5 they are
+  # no longer supported. The BETA resource 'StatefulSets' replaces
+  # them. We can't migrate clients PetSets to
+  # StatefulSets. Additionally, Red Hat has never officially supported
+  # these resource types. Sorry users, but if you were using
+  # unsupported resources from the Kube documentation then we can't
+  # help you at this time.
+  #
+  # Reference: https://bugzilla.redhat.com/show_bug.cgi?id=1428229
+  - name: Check if legacy PetSets exist
+    oc_obj:
+      state: list
+      all_namespaces: true
+      kind: petsets
+    register: l_do_petsets_exist
+
+  - name: FAIL ON Resource migration 'PetSets' unsupported
+    fail:
+      msg: >
+        PetSets are ALPHA status Kube resources and are not supported
+        by Red Hat. Automatically migrating to their BETA replacement,
+        StatefulSets, in 3.5 is not supported. Please see the
+        Kubernetes "Upgrading from PetSets to StatefulSets"
+        documentation for additional information:
+        https://kubernetes.io/docs/tasks/manage-stateful-set/upgrade-pet-set-to-stateful-set/
```

Comment:
I'd just make it clear that the 'OK' button is to delete them all. Perhaps what you have now, a blank line, and the following :
`PetSets must be removed prior to upgrade, If you wish to simply remove all PetSets please run this command as an admin : oc get petsets --all-namespaces -o yaml | oc delete -f - --cascale=false`

I'm just slightly worried about us pointing to kube docs when there may be subtle differences between kube and our implementation and we don't have time to investigate all of them, really they're just going to need to delete them.